workflows/prognostic_c48_run/runtime/radiation_wrapper.py

```diff
@@ -0,0 +1,484 @@
+        dt_atmos: float,
+        forcing_dir: str = "./data/forcing",
+    ) -> None:
+        """Update the radiation driver's time-varying parameters"""
+        if self._rad_config is None:
+            raise ValueError(
+                "Radiation driver not initialized. `.rad_init` must be called "
+                "before `.rad_update`."
+            )
+        # idat is supposed to be model initalization time but is unused w/ current flags
+        idat = np.array(
+            [time.year, time.month, time.day, 0, time.hour, time.minute, time.second, 0]
+        )
+        jdat = np.array(
+            [time.year, time.month, time.day, 0, time.hour, time.minute, time.second, 0]
+        )
+        fhswr = np.array(float(self._rad_config["fhswr"]))
+        dt_atmos = np.array(float(dt_atmos))
+        aerosol_data = getdata.aerosol(forcing_dir)
+        _, solar_data = getdata.astronomy(forcing_dir, self._rad_config["isolar"])
+        gas_data = getdata.gases(forcing_dir, self._rad_config["ictmflg"])
+        self._driver.radupdate(
+            idat,
+            jdat,
+            fhswr,
+            dt_atmos,
+            self._rad_config["lsswr"],
+            aerosol_data["kprfg"],
+            aerosol_data["idxcg"],
+            aerosol_data["cmixg"],
+            aerosol_data["denng"],
+            aerosol_data["cline"],
+            solar_data,
+            gas_data,
+        )
+
+    def rad_compute(
+        self,
+        state: State,
+        tracer_metadata: Mapping[str, Any],
+        time: cftime.DatetimeJulian,
+        dt_atmos: float,
+        rank: int,
+        lookup_dir: str = "./data/lookup",
+    ):
+        if self._rad_config is None:
+            raise ValueError(
+                "Radiation driver not initialized. `.rad_init` must be called "
+                "before `.rad_compute`."
+            )
+        if self._input_model is not None:
+            predictions = predict(self._input_model, state)
+            state = UpdatedState(state, predictions)
+        statein = _statein(state, tracer_metadata, self._rad_config["ivflip"])
+        grid, coords = _grid(state)
+        sfcprop = _sfcprop(state)
+        ncolumns, nz = statein["tgrs"].shape[0], statein["tgrs"].shape[1]
+        model = _model(self._rad_config, tracer_metadata, time, dt_atmos, nz, rank)
+        random_numbers = getdata.random(ncolumns, nz, NGPTSW, NGPTLW)
+        lw_lookup = getdata.lw(lookup_dir)
+        sw_lookup = getdata.sw(lookup_dir)
+        out = self._driver._GFS_radiation_driver(
+            model, statein, sfcprop, grid, random_numbers, lw_lookup, sw_lookup
+        )
+        return _unstack(_rename_out(out), coords)
+
+
+class UpdatedState(DerivedFV3State):
+    def __init__(self, state: State, predictions: State):
+        self._state = state
+        self._predictions = predictions
+
+    def __getitem__(self, key: Hashable) -> xr.DataArray:
+        return self._predictions.get(key, self._state[key])
```

Comment:
This will fail if `key` is not in `self._state` even if it is in `self._predictions`. Is that what you want?